app/models/manageiq/providers/openstack/cloud_manager/auth_key_pair.rb

```diff
@@ -7,7 +7,9 @@ class ManageIQ::Providers::Openstack::CloudManager::AuthKeyPair < ManageIQ::Prov
   def self.raw_create_key_pair(ext_management_system, create_options)
     connection_options = {:service => 'Compute'}
     ext_management_system.with_provider_connection(connection_options) do |service|
-      service.key_pairs.create(create_options)
+      kp = service.key_pairs.create(create_options)
+
+      {:name => kp.name, :fingerprint => kp.fingerprint, :auth_key => kp.private_key}
```

Comment:
I think this is only used to store public keys right?
```suggestion
      {:name => kp.name, :fingerprint => kp.fingerprint, :auth_key => kp.public_key}
```

Reply:
`:auth_key` is used to store the private key, see https://github.com/ManageIQ/manageiq/pull/21759/files#diff-b80f64b28488206bf8d41b9b53fb77a5490267a1a91b5d1e4f0897af88951554L46

Reply:
Thanks for the explanation! You're right - I was looking at the `:public_key` column. Makes sense now, though :)